### PR TITLE
Switch to docker bake

### PR DESCRIPTION
**Pull request by timja**
Test changes are required as part of it, switched to bats-assert as well to make the tests easier to read

### CODE REVIEW

**Comment by timja** [comment]
Tests are failing on alpine java 8, others are passing

File: tests/tests.bats [hunk 1]
```diff
@@ -1,193 +1,139 @@
...
-  [ "$status" = "0" ] && [ "$output" = "f00" ] \
-    || (\
-      echo "status: $status"; \
-      echo "output: $output"; \
-      false \
-    )
+  run_through_ssh "${test_container_name}" echo f00
+  assert_success
+  assert_equal "${output}" "f00"
+
```
Comment (MarkEWaite) [inline tests/tests.bats:66]: Remove trailing white space?
```suggestion

```

File: tests/tests.bats [hunk 2]
```diff
@@ -1,193 +1,139 @@
...
```
Comment (MarkEWaite) [inline tests/tests.bats:80]: Remove trailing white space?
```suggestion

```

File: tests/tests.bats [hunk 3]
```diff
@@ -1,193 +1,139 @@
...
-  [ "$status" = "0" ] && [ "$ret" = "0" ] \
-    || (\
-      echo "status: $status"; \
-      echo "ret: $ret"; \
-      echo "output: $output"; \
-      false \
-    )
+  assert_success
+  assert_output --regexp '^openjdk version \"[[:digit:]]+\.'
+
```
Comment (MarkEWaite) [inline tests/tests.bats:99]: Remove trailing white space?
```suggestion

```

File: tests/tests.bats [hunk 4]
```diff
@@ -1,193 +1,139 @@
...
-  [ "$status" = "0" ] && [ "$output" = "f00" ] \
-    || (\
-      echo "status: $status"; \
-      echo "output: $output"; \
-      false \
-    )
+  run_through_ssh "${test_container_name}" echo f00
+  assert_success
+  assert_equal "${output}" "f00"
+
```
Comment (MarkEWaite) [inline tests/tests.bats:116]: Remove trailing white space?
```suggestion

```

File: tests/tests.bats [hunk 5]
```diff
@@ -1,193 +1,139 @@
...
```
Comment (MarkEWaite) [inline tests/tests.bats:132]: Remove trailing white space?
```suggestion

```

File: tests/tests.bats [hunk 6]
```diff
@@ -196,25 +142,35 @@ DOCKER_PLUGIN_DEFAULT_ARG="/usr/sbin/sshd -D -p 22"
...
+  assert_line --index 0 "${TEST_USER}"
+  run docker exec "${test_container_name}" sh -c "id -g -n ${TEST_USER}"
+  assert_line --index 0 "${TEST_GROUP}"
+  run docker exec "${test_container_name}" sh -c "id -u ${TEST_USER}"
+  assert_line --index 0 "${TEST_UID}"
+  run docker exec "${test_container_name}" sh -c "id -g ${TEST_USER}"
+  assert_line --index 0 "${TEST_GID}"
+  run docker exec "${test_container_name}" sh -c 'stat -c "%U:%G" "${JENKINS_AGENT_HOME}"'
+  assert_line --index 0 "${TEST_USER}:${TEST_GROUP}"
+
```
Comment (MarkEWaite) [inline tests/tests.bats:174]: Remove trailing white space?
```suggestion

```

**Review by MarkEWaite — APPROVED** [review]
Approved once automated tests pass.  They pass on my development environment.  Not sure why they fail on ci.ejnkins.io

**Comment by timja** [comment]
Re-running in case it’s a flake

**Comment by timja** [comment]
Interesting most of them passed on a second run, unfortunately the alpine tests won't run on my machine.

I'll create a VM later to debug

**Comment by timja** [comment]
Passed now, will keep an eye on it

**Pull request merged**